### ISSUE TITLE
Threshold input parameter for indice calculator is broken when using low transfer_limit_Mbytes

### ISSUE THREAD

**Issue by maartenplieger** [cerfacs-globc/icclim#19]
When using transfer_limit_Mbytes = 45, which puts ICCLIM in chunking mode, the custom threshold is not working.

Here is a script to reproduce:

import icclim
import datetime
files=['http://opendap.knmi.nl/knmi/thredds/dodsC/e-obs_0.25regular/tx_0.25deg_reg_v12.0.nc']

dt1=datetime.datetime(1965,1,1)
dt2=datetime.datetime(1965,12,31)
out_f = 'SU_YEAR_E-OBS_observation-opendap_r1i1p1_1965-1986.nc'

icclim.indice(indice_name='SU', in_files=files, var_name='tx', threshold=[-248.15],time_range=[dt1, dt2], transfer_limit_Mbytes=45.0, slice_mode='year', out_file=out_f)

And here is the error:
$ python testicclim.py 
Loading data: chunk 1/2 ...
Loading data: chunk 2/2 ...
Traceback (most recent call last):
  File "testicclim.py", line 17, in <module>
    icclim.indice(indice_name='SU', in_files=files, var_name='tx', threshold=[-248.15],time_range=[dt1, dt2], transfer_limit_Mbytes=45.0, slice_mode='year', out_file=out_f)
  File "/nobackup/users/plieger/c4i_dev/wps/virtenv/lib/python2.7/site-packages/icclim-4.1.1-py2.7.egg/icclim/icclim.py", line 567, in indice
    dict_threshold_indice_arr[t][:, i1_row_current_tile:i2_row_current_tile, i1_col_current_tile:i2_col_current_tile] = indice_arr_current_chunk
KeyError: -248.15

**Comment by marev711** [comment]
I can confirm this error and would guess the fix would be to move the creation of the `dict_threshold_indice_arr` next to its initialization such that it is only carried out once (see patch below). I'm not entirely sure how/when data in this array is supposed to be used so I would have to double check. 
```
         else:
-            dict_threshold_indice_arr = OrderedDict()
             for t in user_thresholds:
                 if chunk_counter == 0:
+                    dict_threshold_indice_arr = OrderedDict()
                     dict_threshold_indice_arr[t] = numpy.zeros( (len(dict_temporal_slices),var_shape1, var_shape2), dtype=in
```

**Comment by maartenplieger** [comment]
Any news on this issue? Need help?


On 05/16/2016 03:43 PM, marev711 wrote:
> I can confirm this error and would guess the fix would be to move the creation of the
> |dict_threshold_indice_arr| next to its initialization such that it is only carried out once (see
> patch below). I'm not entirely sure how/when data in this array is supposed to be used so I would
> have to double check.
>
> |else: - dict_threshold_indice_arr = OrderedDict() for t in user_thresholds: if chunk_counter == 0: +
> dict_threshold_indice_arr = OrderedDict() dict_threshold_indice_arr[t] = numpy.zeros(
> (len(dict_temporal_slices),var_shape1, var_shape2), dtype=in |
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/cerfacs-globc/icclim/issues/19#issuecomment-219427965>
>

-- 
Maarten Plieger
KNMI, R&D Information and Observation Technology, De Bilt
(t) +31 30 2206330

**Comment by pagecp** [comment]
It is now fixed in release 4.2.1. :)
